File: todoList.hs
import System.IO

mainLoop :: String -> IO ()
mainLoop dbFileName = do
    putStrLn "Por favor, ingresa la descripción de la tarea y presiona Enter:"
    taskDescription <- getLine
    appendFile dbFileName (taskDescription ++ "\n")
    putStrLn "\nLista de tareas:"
    showFileContentWithNumbers dbFileName
    mainLoop dbFileName

showFileContentWithNumbers :: String -> IO ()
showFileContentWithNumbers dbFileName = do
    fileContent <- readFile dbFileName
    let tasks = lines fileContent
    mapM_ (\(index, task) -> putStrLn (show index ++ ". " ++ task)) (zip [1..] tasks)

main :: IO ()
main = do
    putStrLn "Bienvenido a tu lista de tareas."
    mainLoop "db.txt"

File: todolist.f
### PseudoCodigo

main = 
    show "Bienvenido a tu lista de tareas." 
    main loop "db.txt"

main loop DB:String = 
    show "Por favor, ingresa la descripción de la tarea y presiona Enter:"
    add to file db ( add new line (get info))
    show "\nLista de tareas:"
    show list with numbers <read db>
    repeat main loop

TaskList = <String> List
show list with numbers <TaskList> = 
>   1. tomar el siguiente elemento del tasklist (for each)
>   2. a ese elemento pongale al principio;
>        2.1 un numero (iniciando en 1 segun su orden)
>        2.2 un punto y un espacio
>   3. muestre el elemento \n
    for each <index, element> in (enumerate tasklist) : 
        text = (index + 1) ++ ". " ++ element
        show text